go-controller/pkg/util/net_unit_test.go

```diff
@@ -278,3 +279,115 @@ func TestMatchIPFamily(t *testing.T) {
 		})
 	}
 }
+
+func TestJoinHostPortInt32(t *testing.T) {
+	tests := []struct {
+		desc    string
+		inpHost string
+		inpPort int32
+		outExp  string
+	}{
+		{
+			desc:    "empty string for host name and a negative port number",
+			inpPort: -25,
+			outExp:  ":-25",
+		},
+		{
+			desc:    "valid non-zero length string for host name",
+			inpHost: "hostname",
+			inpPort: 22,
+			outExp:  "hostname:22",
+		},
+		{
+			desc:    "valid IPv4 address as host name",
+			inpHost: "192.168.1.15",
+			inpPort: 22,
+			outExp:  "192.168.1.15:22",
+		},
+		{
+			desc:    "valid IPv6 address as host name",
+			inpHost: "fd01::1234",
+			inpPort: 22,
+			outExp:  "[fd01::1234]:22",
+		},
+	}
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d:%s", i, tc.desc), func(t *testing.T) {
+			res := JoinHostPortInt32(tc.inpHost, tc.inpPort)
+			t.Log(res)
+			assert.Equal(t, res, tc.outExp)
+		})
+	}
+}
+
+func TestIPAddrToHWAddr(t *testing.T) {
+	tests := []struct {
+		desc  string
+		inpIP net.IP
+	}{
+		// TODO: The function may need to be modified to ensure that the input net.IP passed is not empty.
+		// Passing empty net.IP causes panic
+		/*{
+			desc:	"test empty instance of net.IP",
+			inpIP: net.IP{},
+		},*/
+		{
+			desc:  "test IPv4 instance of net.IP",
+			inpIP: ovntest.MustParseIP("192.168.1.5"),
+		},
+		{
+			desc:  "test IPv6 instance of net.IP",
+			inpIP: ovntest.MustParseIP("fd01::1234"),
+		},
+	}
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d:%s", i, tc.desc), func(t *testing.T) {
+			res := IPAddrToHWAddr(tc.inpIP)
+			t.Log(res)
+			assert.Equal(t, res[0], byte(0x0A))
+			assert.Equal(t, res[1], byte(0x58))
+		})
+	}
+}
+
+func TestJoinIPs(t *testing.T) {
+
+}
+
+func TestJoinIPNets(t *testing.T) {
+	tests := []struct {
+		desc         string
+		inpIPNetList []*net.IPNet
+		inpSeparator string
+		outExp       string
+	}{
+		{
+			desc:         "an empty net.IPNet list with ',' separator",
+			inpSeparator: ",",
+			outExp:       "",
+		},
+		{
+			desc:         "single item in net.IPNet list with `;` separator",
+			inpIPNetList: ovntest.MustParseIPNets("192.168.1.5/24"),
+			inpSeparator: ";",
+			outExp:       "192.168.1.5/24",
+		},
+		{
+			desc:         "two items in net.IPNet list with `;` separator",
+			inpIPNetList: ovntest.MustParseIPNets("192.168.1.5/24", "192.168.1.6/24"),
+			inpSeparator: ";",
+			outExp:       "192.168.1.5/24;192.168.1.6/24",
+		},
+	}
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d:%s", i, tc.desc), func(t *testing.T) {
+			res := JoinIPNets(tc.inpIPNetList, tc.inpSeparator)
+			t.Log(res)
```

Comment:
should we remove this?

Reply:
@girishmg from my past experiences, I found it useful to have logs of the output as well in addition to errors. It makes it easy sometimes when looking at CI results.